internal/store/fleet.go

```diff
@@ -391,3 +397,25 @@ func (s *FleetStore) UpdateConditions(ctx context.Context, orgId uuid.UUID, name
 
 	return err
 }
+
+func (s *FleetStore) UpdateAnnotations(ctx context.Context, orgId uuid.UUID, name string, annotations map[string]string, deleteKeys []string) error {
```

Comment:
question: what is the use-case for `deleteKeys` param on this Update method?

Reply:
If the caller needs to unset an annotation

Reply:
OK so if the annotations map doesn't contain an existing key we don't delete the diff? Feels like an unnessisry step.

Reply:
No, it's not an overwrite API, it's a patch.  We update only the keys that are passed so that we don't accidentally delete keys that a different process is updating.

Reply:
maybe we should document the method?